service/src/test/java/com/epam/rft/atsy/service/impl/StatesServiceImplTest.java

```diff
@@ -0,0 +1,335 @@
+    private static final CandidateEntity CANDIDATE_ENTITY_WITHOUT_APPLICATIONS = CandidateEntity.builder().id(FIRST_ID).build();
+    private static final List<ApplicationEntity> EMPTY_APPLICATION_ENTITY_LIST = Collections.emptyList();
+    private static final List<StateEntity> EMPTY_STATE_ENTITY_LIST = Collections.emptyList();
+    private static final List<StateViewDTO> EMPTY_STATE_VIEW_DTO_LIST = Collections.emptyList();
+
+
+    private Date now;
+    private ApplicationEntity applicationEntity;
+    private CandidateEntity candidateEntity;
+    private PositionEntity positionEntity;
+
+    private StateEntity stateEntity;
+    private StateViewDTO stateViewDTO;
+    private StateDTO stateDTO;
+
+    private List<ApplicationEntity> applicationEntityListWithSingleElement;
+    private List<ApplicationEntity> applicationEntityListWithThreeElements;
+    private List<StateEntity> stateEntityListWithSingleElement;
+    private List<StateEntity> stateEntityListWithThreeElements;
+    private List<StateViewDTO> stateViewDTOListWithSingleElement;
+    private List<StateViewDTO> stateViewDTOListWithThreeElements;
+
+
+    @Mock
+    private ModelMapper modelMapper;
+
+    @Mock
+    private StatesRepository statesRepository;
+    @Mock
+    private ApplicationsRepository applicationsRepository;
+    @Mock
+    private CandidateRepository candidateRepository;
+
+    @InjectMocks
+    private StatesServiceImpl statesService;
+
+
+    @Before
+    public void setUp() {
+        now = new Date();
+        candidateEntity = CandidateEntity.builder().id(FIRST_ID).build();
+        positionEntity = PositionEntity.builder().id(FIRST_ID).name(POSITION_ENTITY_NAME).build();
+        applicationEntity = ApplicationEntity.builder().id(FIRST_ID).candidateEntity(candidateEntity).positionEntity(positionEntity).creationDate(now).build();
+
+        stateEntity = StateEntity.builder().id(FIRST_ID).applicationEntity(applicationEntity).stateType(STATE_TYPE_NAME).creationDate(now).build();
+        stateViewDTO = StateViewDTO.builder().id(FIRST_ID).build();
+        stateDTO = StateDTO.builder().id(FIRST_ID).build();
+
+        applicationEntityListWithSingleElement = Arrays.asList(applicationEntity);
+        applicationEntityListWithThreeElements = Arrays.asList(applicationEntity, applicationEntity, applicationEntity);
+
+        stateEntityListWithSingleElement = Arrays.asList(stateEntity);
+        stateEntityListWithThreeElements = Arrays.asList(stateEntity, stateEntity, stateEntity);
+
+        stateViewDTOListWithSingleElement = Arrays.asList(stateViewDTO);
+        stateViewDTOListWithThreeElements = Arrays.asList(stateViewDTO, stateViewDTO, stateViewDTO);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void getStatesByApplicationIdShouldThrowIllegalArgumentExceptionWhenApplicationIdIsNull() {
+        // Given
+        given(applicationsRepository.findOne(null)).willThrow(IllegalArgumentException.class);
```

Comment:
Do not call repository method, when you can detect if the parameter is wrong.
Use assert in the service implementation for catching invalid parameters.